text/src/autogluon/text/automm/data/collator.py

```diff
@@ -0,0 +1,324 @@
+import math
+import warnings
+import numpy as np
+import torch
+
+__all__ = ["Stack", "Pad", "Tuple", "List", "Dict", "PadImage", "Zip"]
+
+
+def _pad_arrs_to_max_length(arrs, pad_axis, pad_val, round_to=None, max_length=None):
+    """Inner Implementation of the Pad collate
+
+    Arguments:
+        arrs (list)
+        pad_axis (int)
+        pad_val (number)
+        round_to (int, optional). (default: ``None``)
+        max_length (int, optional). (default: ``None``)
+
+    Returns:
+        ret : torch.Tensor
+        original_length : torch.Tensor
+    """
+    if not isinstance(arrs[0], torch.Tensor):
+        arrs = [torch.as_tensor(ele) for ele in arrs]
+
+    original_length = [ele.size(pad_axis) for ele in arrs]
+    max_arr_len = max(original_length)
+
+    if round_to is not None:
+        max_arr_len = round_to * math.ceil(max_arr_len / round_to)
+    elif max_length is not None:
+        if max_length < max_arr_len:
+            raise ValueError(
+                "If max_length is specified, max_length={} must be larger "
+                "than the maximum length {} of the given arrays at axis={}.".format(
+                    max_length, max_arr_len, pad_axis
+                )
```

Comment:
Is there any scenario where we would want to trim to the `max_length` then pad instead of erroring-out?